gmso/core/site.py

```diff
@@ -125,14 +124,29 @@ def atom_type(self, val):
         val = _validate_atom_type(val)
         self._atom_type = val
 
+    @property
+    def position(self):
+        return self._position
+
+    @position.setter
+    def position(self, position):
+        self._position = _validate_position(position)
+
     def __repr__(self):
         return "<Site {}, id {}>".format(self.name, id(self))
 
 
 def _validate_position(position):
+    if position is None:
+        return None
+
     if not isinstance(position, u.unyt_array):
+        try:
```

Comment:
Since we are doing the checking, should we also check/enforce that the given positions have length of 3?

Reply:
I agree with @daico007 , the `position.shape` should be of size `(3,)`.